### DISCUSSION TITLE
Extracting Meta channel from RGB+meta TIFF to single channel gray 16bpp image difficulties

### DISCUSSION THREAD

**Discussion by jaffamuffin** [ImageMagick/ImageMagick#6869] [Help] [answered]
### ImageMagick version

7.1.1-21 Q16-HDRI x64 58d86e4:20231021

### Operating system, version and so on

Microsoft Windows [Version 10.0.19045.3570]

### Description

Hi been at this for a few days now, can't seem to get the right command, i'm sure it must be possible.  (below is a little context, skip to Question for the actual question if tl;dr thanks) 

Background:
I am scanning film with Vuescan, and am producing an RGB 16-bit depth image, (0002.tif) (approx 110MB) and a corresponding 16bpp Gray image (raw0002.tif) (approx 35MB) that contains the infrared data.  I can successfully add this image using imagemagick to the first image as an ExtraSamples gray channel which allows me to further process this data.

`magick 0002.tif ( raw0002.tif -channel-fx "|meta0=>alpha" ) -compose copyopacity -composite -define tiff:alpha=unspecified combined0002.tif
`

Problem:
However I  already scanned a number of images that comprise of the same first RGB image, (0001.tif) but the second image is an RGB+Ir image, i.e. it is precombined. (raw0001.tif) (like my output above) 

But there is colour processing applied to this image by Vuescan so I want to separate the gray/extrasamples (meta0) to a separate file, delete this rawscan, and have the same starting position as above, i.e. 35MB Gray single channel image, and 3 channel RGB image that I can then combine using the command above.

Note, I could copy the channel directly I think, but I want the separate intermediate files so they fit in with the rest of my processes.

Question:

**How do I extract meta0 channel from tiff and save as 16bpp gray single channel image?** 

What I have tried:
Many of the commands I try result in my image being horizontally squashed, all green, all black, always 150+MB (all channels still there...)

I thought I could simply extract the channel data like this: 
`magick raw0001.tif -channel meta0 -separate -colorspace gray out.tif
`
`magick raw0001.tif -channel-fx "meta0=>gray" -colorspace gray  out.tif    
`
`magick raw0001.tif -channel-fx "| meta0=>gray" -colorspace gray  out.tif    
`
`magick raw0001.tif -channel-fx "meta0=>alpha" -extract alpha -colorspace gray  out.tif    
`
all  give me a large 150MB  or 175MB file which still seems to contain RGB channels, ands it's kind of squashed. or the colour is really weird.

Any pointers ? Is it possible.  Also not sure what the pipe does in the fx command.

I have put all images on this we transfer. example of the source image and example of what I'm trying to achieve.

0001.tif - rgb image i want to keep untouched
raw0001.tif - RGB + gray image I want to extract the gray data into a new file of 1 channel 16bpp about 35MB size

0002.tif - example image i want to keep untouched
raw0002.tif - example of accompanying file that I can merge as I wish with 0002.tif, this type of file is what I'm trying to extract/save/convert from raw0001.tif

https://we.tl/t-xsxMYfLZ6x

Many thanks to all

**Comment by snibgo** [comment, marked as answer]
Your zip file is 400 MB. In future, please try to make smaller examples.

In `-channel-fx`, the vertical bar `|` means "take the input from the next image in the current list". In your example command, you have only one input image, so the vertical bar has no effect. This symbol `|` is also used for "pipe" in shells, but it doesn't mean "pipe" here.

When an image contains a meta channel, IM has no operation that will remove that meta channel. So from `raw0001.tif`, which has four channels (RGB and meta), to make an image that has one gray channel only, copied from the meta, we need to start with a grayscale image.

Windows BAT syntax:
```
magick ^
  raw0001.tif ^
  -size %%wx%%h xc:gray(0) ^
  +swap ^
  -channel-fx "| meta0=>gray" ^
  gr.tiff
```
For bash syntax, change `%%` to `%`, an d `^` to `\`.

What does this do?

line 1: read your tiff file.
line 2: make an image the same size, with just a gray channel.
line 3: swap the order of the images
line 4: modify the gray image by copying the meta0 channel from the second image to the gray channel.
line 5: write this gray image to a file.

Does that help?

**Reply by jaffamuffin** [reply]
Hello snibgo

thanks for the message

Sorry about the large files..  I have put a smaller crop here (8MB) ( magick raw0001.tif -crop 1000x1000+3000+1500 +repage raw_crop.tif )
https://we.tl/t-0C4XRtxUeE

But, your command works perfectly. Thanks for the help and the clear explanation.
